docs/docs/icicle/polynomials/overview.md

```diff
@@ -0,0 +1,364 @@
+
+### Templating in the Polynomial API
+
+The Polynomial API is designed with a templated structure to accommodate different data types for coefficients, the domain, and images. This flexibility allows the API to be adapted for various computational needs and types of data.
+
+```cpp
+template <typename Coeff, typename Domain = Coeff, typename Image = Coeff>
+class Polynomial {
+    // Polynomial class definition
+}
+```
+In this template:
+
+- **Coeff**: Represents the type of the coefficients of the polynomial.
+- **Domain**: Specifies the type for the input values over which the polynomial is evaluated. By default, it is the same as the type of the coefficients but can be specified separately to accommodate different computational contexts.
+- **Image**: Defines the type of the output values of the polynomial. This is typically the same as the coefficients.
+
+#### Default instantiation
+```cpp
+extern template class Polynomial<scalar_t>;
+```
+
+#### Extended use cases
+The templated nature of the Polynomial API also supports more complex scenarios. For example, coefficients and images could be points on an elliptic curve (EC points), which are useful in cryptographic applications and advanced algebraic structures. This approach allows the API to be extended easily to support new algebraic constructions without modifying the core implementation.
+
+### Supported Operations
+The Polynomial class encapsulates a polynomial, providing a variety of operations:
+- **Construction**: Create polynomials from coefficients or evaluations on roots-of-unity domains.
+- **Arithmetic Operations**: Perform addition, subtraction, multiplication, and division.
+- **Evaluation**: Directly evaluate polynomials at specific points or across a domain.
+- **Manipulation**: Features like slicing polynomials, adding or subtracting monomials inplace, and computing polynomial degrees.
+- **Memory Access**: Access internal states or obtain device-memory views of polynomials.
+
+## Usage
+
+This section outlines how to use the Polynomial API in C++. Bindings for Rust and Go are detailed under the Bindings sections.
+
+### Backend Initialization
+Initialization with an appropriate factory is required to configure the computational context and backend.
+
+```cpp
+#include "polynomials/polynomials.h"
+#include "polynomials/cuda_backend/polynomial_cuda_backend.cuh"
+
+// Initialize with a CUDA backend
+Polynomial::initialize(std::make_shared<CUDAPolynomialFactory>());
+```
+
+**Note**: Icicle is built to a library per field/curve. Initialization must be done per library. That is, applications linking to multiple curves/fields should do it per curve/field.
```

Comment:
Will it be rendered as a panel? If not - check this
https://docusaurus.io/docs/markdown-features/admonitions